### PR TITLE
[WIN32SS][FONT] Use HHEA table metrics if (fsSelection & 0x80)

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
JIRA issue: [CORE-15166](https://jira.reactos.org/browse/CORE-15166)
See also: https://docs.microsoft.com/en-us/typography/opentype/spec/os2#fsselection

## Proposed changes
- If (pOS2->fsSelection & 0x80) is true, then use pHori->Ascender and pHori->Descender (it's recommended).

### CODE REVIEW

**Comment by katahiromz** [comment]
This PR will fix tiny glyph rendering.

File: win32ss/gdi/ntgdi/freetype.c [hunk 1]
```diff
@@ -3317,11 +3317,12 @@ IntRequestFontSize(PDC dc, PFONTGDI FontGDI, LONG lfWidth, LONG lfHeight)
         return FT_Request_Size(face, &req);
     }
 
+    /* See also: https://docs.microsoft.com/en-us/typography/opentype/spec/os2#fsselection */
     if (lfHeight > 0)
     {
         /* case (A): lfHeight is positive */
         Sum = pOS2->usWinAscent + pOS2->usWinDescent;
-        if (Sum == 0)
+        if (Sum == 0 || (pOS2->fsSelection & 0x80))
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/freetype.c]: Can you define a `FM_SEL_USE_TYPO_METRICS` flag with this value of 0x80 in the code?
I based the name on the flag value "name" given in your MSDN link and from the documented fsSelection flag names from https://docs.microsoft.com/en-us/windows/desktop/api/winddi/ns-winddi-_fontdiff .

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: OK.

**Comment by learn-more** [comment]
This is how to run a build from a PR:
(example)[https://i.imgur.com/5Ho6k2k.png]

It is running as: https://build.reactos.org/builders/Build%20GCCLin_x86/builds/23068

**Comment by katahiromz** [comment]
https://reactos.org/testman/compare.php?ids=23068,23078

**Pull request merged**